frontend/components/wallet/WalletDepositWithdrawBlock.vue

```diff
@@ -0,0 +1,135 @@
+                    the “mainnet” network is
+                    <FormatAddress type="address" value="0x6b175474e89094c44da98b954eedeac495271d0f" shorten />)
+                </li>
+            </ul>
+        </TextBlock>
+        <div>
+            <RadioGroup
+                :disabled="isLoading"
+                :value="selectedMethod"
+                button-style="solid"
+                class="flex items-center w-full"
+                @change="handleMethodChange"
+            >
+                <RadioButton value="deposit" class="w-full text-center"> Deposit </RadioButton>
+                <RadioButton value="withdraw" class="w-full text-center"> Withdraw </RadioButton>
+            </RadioGroup>
+        </div>
+
+        <WalletDAIInput
+            v-if="selectedMethod === 'deposit'"
+            :value="depositAmount"
+            :disabled="isLoading"
+            :max="maxDeposit"
+            @update="setDepositAmount"
+        />
+        <WalletDAIInput
+            v-else
+            :value="withDrawAmount"
+            :disabled="isLoading"
+            :max="maxWithdraw"
+            @update="setWithdrawAmount"
+        />
+
+        <BaseButton type="primary" class="capitalize" :disabled="isLoading || !canSubmit" @click="submit">
+            {{ selectedMethod }}
+        </BaseButton>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Radio } from 'ant-design-vue';
+import BigNumber from 'bignumber.js';
+import TextBlock from '../common/TextBlock.vue';
+import FormatAddress from '../utils/FormatAddress.vue';
+import BaseButton from '../common/BaseButton.vue';
+import WalletDAIInput from './WalletDAIInput.vue';
+
+export default Vue.extend({
+    name: 'WalletDepositWithdrawBlock',
+    components: {
+        WalletDAIInput,
+        BaseButton,
+        FormatAddress,
+        TextBlock,
+        RadioGroup: Radio.Group,
+        RadioButton: Radio.Button,
+    },
+    props: {
+        isLoading: {
+            type: Boolean,
+            default: false,
+        },
+        maxDeposit: {
+            type: Number,
+            default: 0,
+        },
+        maxWithdraw: {
+            type: Number,
+            default: 0,
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+    },
+    data() {
+        return {
+            selectedMethod: 'deposit',
+            depositAmount: this.maxDeposit,
+            withDrawAmount: this.maxWithdraw,
+        };
+    },
+    computed: {
+        canSubmit(): boolean {
+            if (this.selectedMethod === 'deposit') {
+                return this.depositAmount > 0 && this.depositAmount <= this.maxDeposit;
+            }
+            if (this.selectedMethod === 'withdraw') {
+                return this.withDrawAmount > 0 && this.withDrawAmount <= this.maxWithdraw;
+            }
+            return false;
```

Comment:
After replacing the input, this instead should check if the emitted value is NaN, since the validation logic already exists inside input